pkg/controllers/pushsecret/pushsecret_controller.go

```diff
@@ -137,6 +139,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}
 	syncedSecrets, err := r.PushSecretToProviders(ctx, secretStores, ps, secret, mgr)
 	if err != nil {
+		if err.Error() == errGetSecretClient{
```

Comment:
String matching is dodgy, this will break if someone changes the error message in the path. The error `const`  also contains `test-store` in the error string which is compared and only matches when the secret store is named `test-store`.

Please use `errors.Is()` and return a custom error type so we can test for this specific error: 

https://github.com/external-secrets/external-secrets/blob/main/pkg/controllers/secretstore/client_manager.go#L116

The error also seems to be wrapped already.